certification/artifacts/artifacts.go

```diff
@@ -20,59 +25,40 @@ const DefaultArtifactsDir = "artifacts"
 
 func init() {
 	// set the singleton to the default value.
-	ads = DefaultArtifactsDir
+	cwd, _ := os.Getwd()
+	ads = filepath.Join(cwd, DefaultArtifactsDir)
 }
 
 // SetDir sets the package level artifacts directory. This
 // can be a relative path or a full path.
 func SetDir(s string) {
-	ads = s
+	fullPath := s
+	if !strings.HasPrefix(s, "/") {
+		cwd, _ := os.Getwd()
+		fullPath = filepath.Join(cwd, s)
+	}
+	ads = fullPath
 }
 
 // Reset restores the default value for the Artifacts Directory.
 func Reset() {
-	ads = DefaultArtifactsDir
+	cwd, _ := os.Getwd()
+	ads = filepath.Join(cwd, DefaultArtifactsDir)
 }
 
 // WriteFile will write contents of the string to a file in
 // the artifacts directory. It will create the artifacts dir
 // if necessary.
 // Returns the full path (including the artifacts dir)
-func WriteFile(filename, contents string) (string, error) {
-	artifactDir := ads
-	_, err := createArtifactsDir(artifactDir)
-	if err != nil {
-		// Fatal does an os.Exit. If we can't create the artifacts directory,
-		// we can't continue.
-		log.Fatal(fmt.Errorf("could not create artifact path: %v", err))
-	}
+func WriteFile(filename string, contents io.Reader) (string, error) {
 	fullFilePath := filepath.Join(Path(), filename)
 
-	if err := os.WriteFile(fullFilePath, []byte(contents), 0o644); err != nil {
+	if err := afero.SafeWriteReader(appFS, fullFilePath, contents); err != nil {
 		return fullFilePath, fmt.Errorf("could not write file to artifacts diretory: %v", err)
```

Comment:
Unrelated, but since we're here:

```suggestion
		return fullFilePath, fmt.Errorf("could not write file to artifacts directory: %v", err)
```

Reply:
Surprised the linter didn't catch that.